xdsl/dialects/pdl_interp.py

```diff
@@ -0,0 +1,664 @@
+from __future__ import annotations
+
+from collections.abc import Iterable, Sequence, Set
+from dataclasses import dataclass
+from typing import ClassVar
+
+from xdsl.dialects.builtin import (
+    I16,
+    I32,
+    ArrayAttr,
+    ContainerOf,
+    DictionaryAttr,
+    Float16Type,
+    Float32Type,
+    Float64Type,
+    FunctionType,
+    IndexType,
+    IntegerAttr,
+    IntegerType,
+    StringAttr,
+    SymbolRefAttr,
+    UnitAttr,
+)
+from xdsl.dialects.pdl import (
+    AnyPDLTypeConstr,
+    AttributeType,
+    OperationType,
+    RangeType,
+    TypeType,
+    ValueType,
+)
+from xdsl.ir import Attribute, Block, Dialect, Operation, Region, SSAValue
+from xdsl.irdl import (
+    AnyAttr,
+    AnyOf,
+    AttrSizedOperandSegments,
+    ConstraintContext,
+    GenericAttrConstraint,
+    IRDLOperation,
+    VarConstraint,
+    base,
+    irdl_op_definition,
+    operand_def,
+    opt_prop_def,
+    prop_def,
+    region_def,
+    result_def,
+    successor_def,
+    traits_def,
+    var_operand_def,
+)
+from xdsl.traits import (
+    CallableOpInterface,
+    IsolatedFromAbove,
+    IsTerminator,
+    SymbolOpInterface,
+)
+from xdsl.utils.exceptions import VerifyException
+from xdsl.utils.hints import isa
+
+boolLike = ContainerOf(IntegerType(1))
+signlessIntegerLike = ContainerOf(AnyOf([IntegerType, IndexType]))
+floatingPointLike = ContainerOf(AnyOf([Float16Type, Float32Type, Float64Type]))
+
+
+@irdl_op_definition
+class GetOperandOp(IRDLOperation):
+    """
+    https://mlir.llvm.org/docs/Dialects/PDLInterpOps/#pdl_interpget_operand-pdl_interpgetoperandop
```

Comment:
See external [documentation](https://mlir.llvm.org/docs/Dialects/PDLInterpOps/#pdl_interpget_operand-pdl_interpgetoperandop).

Similarly, below. A micro-PR fixing the documentation would be good.

Reply:
Yeah I missed this, hopefully easy enough to do with a find/replace script.

Reply:
Yeah, this is in order to have this as a "live" URL in our readthedocs documentation :)

no sweat